cli/internal/context/context.go

```diff
@@ -357,18 +357,13 @@ func (c *Context) parsePackageJSON(repoRoot turbopath.AbsoluteSystemPath, pkgJSO
 	defer c.mutex.Unlock()
 
 	if pkgJSONPath.FileExists() {
-		pkg, err := fs.ReadPackageJSON(pkgJSONPath)
+		pkg, err := fs.ReadPackageJSON(repoRoot, pkgJSONPath)
 		if err != nil {
 			return fmt.Errorf("parsing %s: %w", pkgJSONPath, err)
 		}
 
-		relativePkgJSONPath, err := repoRoot.PathTo(pkgJSONPath)
-		if err != nil {
-			return err
-		}
+		// TODO(mehulkar): Figure out if adding duplicate names to the WorkspaceGraph is a problem.
 		c.WorkspaceGraph.Add(pkg.Name)
-		pkg.PackageJSONPath = turbopath.AnchoredSystemPathFromUpstream(relativePkgJSONPath)
-		pkg.Dir = turbopath.AnchoredSystemPathFromUpstream(filepath.Dir(relativePkgJSONPath))
```

Comment:
Assign this during unmarshalling, so the root package.json can also get these props set (we need the Dir so we can look up turbo config for the root workspace)